projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -699,6 +699,56 @@ public static function enqueue_block_editor_assets() {
 		);
 	}
 
+	/**
+	 * Add the Gutenberg editor stylesheet to iframed editors, such as the site editor,
+	 * which don't have access to stylesheets added with `wp_enqueue_style`.
+	 *
+	 * This workaround is only supposed to be used by WordPress.com Simple sites.
```

Comment:
It's quite possible this is needed when other plugins are used that do asset optimization in specific ways, so I wouldn't say it's only for Simple Sites.